File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/MythicKeystone/MythicKeystoneModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.MythicKeystone;

public class MythicKeystoneModel
{
    [JsonPropertyName("current_period")]
    public MythicKeystoneCurrentPeriodModel CurrentPeriod { get; set; }

    [JsonPropertyName("seasons")]
    public MythicKeystoneSeasonModel[] Seasons { get; set; }

    [JsonPropertyName("character")]
    public DungeonCharacterModel Character { get; set; }

    [JsonPropertyName("current_mythic_rating")]
    public MythicKeystoneRaitingModel CurrentMythicRating { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/mythicKeystone/MythicKeystoneModel.ts
import type { DungeonCharacterModel } from "./DungeonCharacterModel";
import type { MythicKeystoneCurrentPeriodModel } from "./MythicKeystoneCurrentPeriodModel";
import type { MythicKeystoneRaitingModel } from "./MythicKeystoneRaitingModel";
import type { MythicKeystoneSeasonModel } from "./MythicKeystoneSeasonModel";

export type MythicKeystoneModel = {
    currentPeriod: MythicKeystoneCurrentPeriodModel;
    seasons: MythicKeystoneSeasonModel[];
    character: DungeonCharacterModel;
    currentMythicRating: MythicKeystoneRaitingModel;
}